cluster/dragon/dragon.go

```diff
@@ -976,34 +985,47 @@ func (d *Dragon) tableExists(queryExec common.SimpleQueryExec, id uint64) (bool,
 	return rows.RowCount() != 0, nil
 }
 
-// Currently the number of shards in the cluster is fixed. We must make sure the user doesn't change the number of shards
-// in the config after the cluster has been created. So we store the number in the database and check
-func (d *Dragon) checkConstantShards(expectedShards int) error {
-	log.Debugf("Checking constant shards: %d", expectedShards)
+func (d *Dragon) GetConfigProperty(property string) ([]byte, error) {
+	key := table.EncodeTableKeyPrefix(common.LocalConfigTableID, 0, 16)
+	propKey := []byte(property)
+	key = common.AppendUint32ToBufferBE(key, uint32(len(propKey)))
+	key = append(key, propKey...)
+	return d.LocalGet(key)
+}
+
+func (d *Dragon) SetConfigProperty(property string, v []byte) error {
 	key := table.EncodeTableKeyPrefix(common.LocalConfigTableID, 0, 16)
-	propKey := []byte("num-shards")
+	propKey := []byte(property)
 	key = common.AppendUint32ToBufferBE(key, uint32(len(propKey)))
 	key = append(key, propKey...)
-	v, err := d.LocalGet(key)
+	batch := d.pebble.NewBatch()
+	if err := batch.Set(key, v, nil); err != nil {
+		return errors.WithStack(err)
+	}
+	if err := d.pebble.Apply(batch, syncWriteOptions); err != nil {
+		return errors.WithStack(err)
+	}
+	return nil
+}
+
+func (d *Dragon) checkConstantProperty(property string, expectedValue int) error {
+	log.Debug("Checking constant %s: %d", property, expectedValue)
+	v, err := d.GetConfigProperty(property)
 	if err != nil {
 		return err
 	}
 	if v == nil {
-		log.Debug("New cluster - no value for num-shards in storage, persisting it")
-		// New cluster - persist the value
-		v = common.AppendUint32ToBufferBE([]byte{}, uint32(expectedShards))
-		batch := d.pebble.NewBatch()
-		if err := batch.Set(key, v, nil); err != nil {
-			return errors.WithStack(err)
-		}
-		if err := d.pebble.Apply(batch, syncWriteOptions); err != nil {
-			return errors.WithStack(err)
+		log.Debugf("New cluster - no value for %s in stroage, persisting it", property)
+		v = common.AppendUint32ToBufferBE([]byte{}, uint32(expectedValue))
```

Comment:
We typically use LittleEndian for values and BigEndian for keys

Reply:
I've made the changes.